lib/Entities/NewsletterPostEntity.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace MailPoet\Entities;
+
+use MailPoet\Doctrine\EntityTraits\AutoincrementedIdTrait;
+use MailPoet\Doctrine\EntityTraits\CreatedAtTrait;
+use MailPoet\Doctrine\EntityTraits\SafeToOneAssociationLoadTrait;
+use MailPoet\Doctrine\EntityTraits\UpdatedAtTrait;
+use MailPoetVendor\Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Entity()
+ * @ORM\Table(name="newsletter_posts")
+ */
+class NewsletterPostEntity {
+  use AutoincrementedIdTrait;
+  use CreatedAtTrait;
+  use UpdatedAtTrait;
+  use SafeToOneAssociationLoadTrait;
+
+  /**
+   * @ORM\ManyToOne(targetEntity="MailPoet\Entities\NewsletterEntity", inversedBy="newsletterSegments")
+   * @var NewsletterEntity
```

Comment:
I guess you need to annotate this `@var NewsletterEntity|null`. After https://github.com/mailpoet/mailpoet/pull/2814 was merged this would fail PHPStan, I guess.

(By default phpstan-doctrine thinks the association is nullable. In DB it is not, which would be fixable by adding `@JoinColumn(nullable=false)` but since we use the `SafeToOneAssociationTrait` we're actually making it nullable indeed.)